.github/workflows/test-full-sync.yml

```diff
@@ -135,6 +135,17 @@ jobs:
         with:
           credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
 
+      # Check if our destination compute instance exists and delete it
+      - name: Delete existing instance with same SHA
+        id: delete-old-instance
+        run: |
+          INSTANCE=$(gcloud compute instances list --filter=full-sync-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} --format='value(NAME)')
+          if [ -z "${INSTANCE}" ]; then
+            echo "No instance to delete"
+          else
+            gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
+          fi
+
       # Creates Compute Engine virtual machine instance w/ disks
       - name: Create GCP compute instance
         id: create-instance
```

Comment:
Should this also depend on `delete-old-instance` succeeding?

Reply:
If `delete-old-instance` success, this step will follow. If it fails (as there's no `if` in `create-instance`) `delete-old-instance` won't run. Other steps are just being extra verbose, I'll fix that in another PR (globally).

Reply:
Gotcha, thanks!